mindmeld/augmentation.py

```diff
@@ -0,0 +1,332 @@
+                self._resource_loader.filter_file_paths(
+                    compiled_pattern=compiled_pattern, file_paths=all_file_paths
+                )
+            )
+        return filtered_paths
+
+    def _write_to_file(self, path, augmented_queries, suffix):
+        """Writes augmented queries to a new file in the path.
+
+        Args:
+            path (str): File path to the original file.
+            augmented_queries (list): List of augmented queries returned by class.
+        """
+        write_path = path.strip(".txt") + suffix
+
+        with open(write_path, "w") as outfile:
+            for query in augmented_queries:
+                outfile.write(query.rstrip() + "\n")
+
+
+class EnglishParaphraser(Augmentor):
+    """Paraphraser class for generating English paraphrases."""
+
+    def __init__(
+        self, batch_size, lang, paths, path_suffix, resource_loader
+    ):
+        """Initializes an English paraphraser.
+
+        Args:
+            batch_size (int): Batch size for batch processing.
+            lang (str): The lang code for paraphrasing.
+            paths (list): Path rules for fetching relevant files to Paraphrase.
+            path_suffix (str): Suffix to be added to new augmented files.
+            resource_loader (object): Resource Loader object for the application.
+        """
+        super().__init__(
+            lang=lang,
+            paths=paths,
+            path_suffix=path_suffix,
+            resource_loader=resource_loader,
+        )
+
+        PegasusTokenizer = _get_module_or_attr("transformers", "PegasusTokenizer")
+        PegasusForConditionalGeneration = _get_module_or_attr(
+            "transformers", "PegasusForConditionalGeneration"
+        )
+
+        model_name = "tuner007/pegasus_paraphrase"
+        torch_device = (
+            "cuda" if _get_module_or_attr("torch.cuda", "is_available")() else "cpu"
+        )
+        self.tokenizer = PegasusTokenizer.from_pretrained(model_name)
+        self.model = PegasusForConditionalGeneration.from_pretrained(
+            model_name, force_download=False
+        ).to(torch_device)
+
+        # Update default params with user model config
+        self.batch_size = batch_size
+
+        self.default_paraphraser_model_params = {
+            "max_length": 60,
+            "num_beams": 10,
+            "num_return_sequences": 10,
+            "temperature": 1.5,
+        }
+
+    def _get_response(self, queries):
+        """Generates paraphrase responses for given query.
+
+        Args:
+            queries (list(str)): List of application queries.
+
+        Return:
+            paraphrases (list(str)): List of paraphrased queries.
+        """
+        all_generated_queries = []
+        for pos in range(0, len(queries), self.batch_size):
+            batch = self.tokenizer.prepare_seq2seq_batch(
+                queries[pos : pos + self.batch_size],
+                truncation=True,
+                padding="longest",
```

Comment:
this should also be `self.default_paraphraser_model_params["padding"]`

Reply:
Tokenizer and model params bunched together?

Reply:
made a ``default_tokenizer_params`` separately.